### PR TITLE
Enable TLS by default in shipper output

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?

closes https://github.com/elastic/beats/issues/34321

This changes the default config for the shipper output so TLS is enabled by default, which is what elastic-agent and shipper expect.

## How to test:

- follow the setup instructions here: https://github.com/elastic/elastic-agent-shipper/pull/185
- If setup correctly, the shipper will _not_ be able to connect, as there's still missing pieces in the shipper's TLS. You can verify that the issue is at the TLS level by adding the following to the `shipper.spec.yml` and looking in the logs:
```
      env:
        - name: GRPC_GO_LOG_VERBOSITY_LEVEL
          value: 99
        - name: GRPC_GO_LOG_SEVERITY_LEVEL
          value: info
        - name: GRPC_TRACE
          value: all
```

### CODE REVIEW

**Review by rdner — COMMENTED** [review]
The description needs manual testing steps, how can we check that the shipper is actually running with TLS? The config option might be as well ignored.

**Comment by fearful-symmetry** [comment]
@rdner so, I'm sort of going off of @blakerouse 's comment in the [issue](https://github.com/elastic/beats/issues/34321), which is that TLS should be enabled, unless the agent explicitly disabled it. Unless I'm misunderstanding your question?

**Comment by rdner** [comment]
@fearful-symmetry my point was rather that would be nice to have steps in the description to test the change. Just to make sure the configuration parameter is not lost in some propagation.

If the TLS on is the default, then we need to verify that the flag actually disables it when set to `false`.

I just don't see anything in this PR that checks or verifies that, unless I've missed it.

**Comment by fearful-symmetry** [comment]
@rdner 
> my point was rather that would be nice to have steps in the description to test the change. Just to make sure the configuration parameter is not lost in some propagation.

Ah, sorry, didn't understand what you meant, sorry. I can try to add some testing instructions, but at this point, there's so many bugs and in-flight PRs needed for the shipper that I don't think it's particularly easy to test. 

I agree though, gonna at least see if  I can add some go tests.

**Comment by fearful-symmetry** [comment]
Alright, added a test and a some more to the description.

File: libbeat/outputs/shipper/config.go [hunk 1]
```diff
@@ -53,6 +53,7 @@ type Config struct {
 func defaultConfig() Config {
 	enabled := true
 	return Config{
+		// agent will expect that TLS is enabled by default, will disable explicitly
```
Comment (rdner) [inline libbeat/outputs/shipper/config.go:56]: > will disable explicitly

Looks confusing because of the `enabled := true` above. What does it mean?

Comment (fearful-symmetry) [reply libbeat/outputs/shipper/config.go:56]: Ah, sorry, it should say "the agent will disabled it explicitly"

**Comment by fearful-symmetry** [comment]
Brief update: Since we currently don't have a timeline for when we want to make changes to the rest of the TLS stack, I'm thinking of leaving this PR unmerged until https://github.com/elastic/elastic-agent-shipper/issues/224 is dealt with, as otherwise we'd need to come up with another workaround to disable TLS until we fix that.

**Comment by jlind23** [comment]
@leehinman @pierrehilbert @fearful-symmetry as https://github.com/elastic/elastic-agent-shipper/issues/224 is still not planned yet, can I move https://github.com/elastic/beats/issues/34321 out fo the current sprint as otherwise it will stay unmerged?

**Comment by pierrehilbert** [comment]
I was expecting to discuss about this on tomorrow in the shipper area but yes I think we can move this one out of the sprint

**Pull request closed without merging**